### PR TITLE
[VEN-2166]: Dynamically Set Addresses for XVS and XVSVToken 

### PR DESCRIPTION
**Pull request by web3rover**
## Description

Resolves #

## Checklist

- [ ] I have updated the documentation to account for the changes in the code.
- [ ] If I added new functionality, I added tests covering it.
- [ ] If I fixed a bug, I added a test preventing this bug from silently reappearing again.
- [ ] My contribution follows [Venus contribution guidelines](docs/CONTRIBUTING.md).

### CODE REVIEW

File: tests/hardhat/Fork/comptrollerXVSAddressesUpgrade.ts [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+      expect(await rewardFacet.getXVSAddress()).to.be.equal("0xcF6BB5389c92Bdda8a3747Ddb454cB7a64626C63");
+
+      const XVS: BEP20 = await ethers.getContractAt(
+        "contracts/test/BEP20.sol:BEP20",
+        "0xcF6BB5389c92Bdda8a3747Ddb454cB7a64626C63",
+      );
+
+      const userAccount = "0xd8F32fe3eeA457bBa2e5e2E1533Ad5f34f591458";
+      const previouseXVSBalance = await XVS.balanceOf(userAccount);
+      console.log();
```
Comment (chechu) [inline tests/hardhat/Fork/comptrollerXVSAddressesUpgrade.ts:126]:
```suggestion
```

**Review by chechu — APPROVED** [review]

File: tests/hardhat/Fork/comptrollerXVSAddressesUpgrade.ts [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+
+const forking = (blockNumber: number, fn: () => void) => {
+  describe(`Reward and Setter facet upgrade check At block #${blockNumber}`, () => {
+    before(async () => {
+      await setForkBlock(blockNumber);
+    });
+    fn();
+  });
+};
+
```
Comment (Debugger022) [inline tests/hardhat/Fork/comptrollerXVSAddressesUpgrade.ts:40]: We can import these functions from [utils.ts](https://github.com/VenusProtocol/venus-protocol/blob/7d3e10178e51982a7474a2831fc38e8f89a0faeb/tests/hardhat/Fork/utils.ts).

File: contracts/Comptroller/Diamond/facets/SetterFacet.sol [hunk 3]
```diff
@@ -87,6 +87,12 @@ contract SetterFacet is ISetterFacet, FacetBase {
     /// @notice Emitted when forced liquidation is enabled or disabled for a user borrowing in a market
     event IsForcedLiquidationEnabledForUserUpdated(address indexed borrower, address indexed vToken, bool enable);
 
+    /// @notice Emitted when XVS token address is changed
+    event NewXVSToken(address oldXVS, address newXVS);
+
+    /// @notice Emitted when XVS vToken address is changed
+    event NewXVSVToken(address oldXVSVToken, address newXVSVToken);
+
```
Comment (Debugger022) [inline contracts/Comptroller/Diamond/facets/SetterFacet.sol:95]: We can make these events parameters `indexed`.

**Review by GitGuru7 — APPROVED** [review]

File: deployments/opbnbmainnet.json
Comment (kkirka) [inline deployments/opbnbmainnet.json:1]: Deployment files don't seem related to this PR

File: tests/hardhat/Fork/comptrollerXVSAddressesUpgrade.ts [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+import chai from "chai";
+import { ethers } from "hardhat";
+
+import { BEP20, Diamond, RewardFacet, SetterFacet } from "../../../typechain";
+import { forking } from "./utils";
+
+const { expect } = chai;
+chai.use(smock.matchers);
+
+const Owner = "0x939bd8d64c0a9583a7dcea9933f7b21697ab6396";
```
Comment (kkirka) [inline tests/hardhat/Fork/comptrollerXVSAddressesUpgrade.ts:13]: Capitalized names usually refer to classes and types, I wouldn't expect it to be a string constant

```suggestion
const OWNER = "0x939bd8d64c0a9583a7dcea9933f7b21697ab6396";
```

File: tests/hardhat/Fork/comptrollerXVSAddressesUpgrade.ts [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+      const rewardFacet = await ethers.getContractAt("RewardFacet", diamond.address);
+
+      expect(await rewardFacet.getXVSVTokenAddress()).to.be.equal("0x151B1e2635A717bcDc836ECd6FbB62B674FE3E1D");
+      expect(await rewardFacet.getXVSAddress()).to.be.equal("0xcF6BB5389c92Bdda8a3747Ddb454cB7a64626C63");
+    });
+
+    it("upgrade checks", async () => {
+      const oldSetterFacet = await ethers.getContractAt("SetterFacet", diamond.address);
+      await expect(
+        oldSetterFacet.connect(owner)._setXVSToken("0x151B1e2635A717bcDc836ECd6FbB62B674FE3E1D"),
```
Comment (kkirka) [inline tests/hardhat/Fork/comptrollerXVSAddressesUpgrade.ts:41]:
```suggestion
        oldSetterFacet.connect(owner)._setXVSToken("0xcF6BB5389c92Bdda8a3747Ddb454cB7a64626C63"),
```

File: tests/hardhat/Fork/comptrollerXVSAddressesUpgrade.ts [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+          facetAddress: newSetterFacet.address,
+          action: 1,
+          functionSelectors: existingSetterFacetFunctions,
+        },
+        {
+          facetAddress: newRewardFacet.address,
+          action: 1,
+          functionSelectors: existingRewardFacetFunctions,
+        },
+      ];
```
Comment (kkirka) [inline tests/hardhat/Fork/comptrollerXVSAddressesUpgrade.ts:81]: Maybe move the numbers to an enum? e.g.

```ts
enum FacetCutAction {
     Add,
     Replace,
     Remove
}
```

```ts
const cut = [
  {
    facetAddress: newSetterFacet.address,
    action: FacetCutAction.Add,
    functionSelectors: [addXVSAddressFunctionSignature, addXVSVTokenAddressFunctionSignature],
  },
  {
    facetAddress: newSetterFacet.address,
    action: FacetCutAction.Replace,
    functionSelectors: existingSetterFacetFunctions,
  },
  {
    facetAddress: newRewardFacet.address,
    action: FacetCutAction.Replace,
    functionSelectors: existingRewardFacetFunctions,
  },
];
```

I'd even store this enum in some utility file so that we could reuse it in tests

**Review by kkirka — APPROVED** [review]

**Review by chechu — APPROVED** [review]
